### PR TITLE
Add slits to the IDF of D17, ILL

### LINKED ISSUE #21781: Add slits to the ILL D17 IDF
The ILL reflectometer D17 IDF requires two new static components 'slit2' and 'slit3' for resolution computation.

(The Figaro instrument already has slits defined)

### PR DESCRIPTION
**Pull request by vreimund**
The ILL reflectometer D17 needs two new components, slit2 and slit3. For future resolution analysis.

No risk.

A description of the instrument:

https://www.ill.eu/instruments-support/instruments-groups/instruments/d17/characteristics/

When loading unit test data for ILL/D17, the `SampleLogs` contain a field `Distance.S2toSample` and `Distance.S3toSample` (unluckily without unit attached). Those values are manual inputs from the instrument responsible. Slits position are subject to manual instrument modification(s).

The IDF now contains reasonable default values which may differ from those values. It will be up to the loader `LoadILLReflectometry` to update the slit positions according to the actual experiment.

I decided to take a constant slit distance of 3.4 m into account and a slit2 z-value of -3.605. The slit3 values seem to vary among different files for different experiments.

For generating the IDF: https://github.com/mantidproject/mantidgeometry/pull/101/

Fixes #21781.

Since changes are made for future resolution analysis:

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by NickDraper** [comment]
@reimundILL  Are you happy that these changes will go out immediately once they are merged?  i.e. without waiting for the release?

**Comment by reimundILL** [comment]
Yes. We have a critical test scientist involved who worries about (not understood) changes in IDFs - I called him this Morning and let him know about this change.

File: instrument/D17_Definition.xml
Comment (NickDraper) [inline instrument/D17_Definition.xml]: I don't think this line is needed, There is no component of type Slit.

**Review by NickDraper — CHANGES_REQUESTED** [review]

File: instrument/D17_Definition.xml
Comment (reimundILL) [reply instrument/D17_Definition.xml]: This is true. Lines 87 and 91 need Slit removed. This applies as well for the Figaro IDF, where I used the same function.

**Review by NickDraper — APPROVED** [review]

**Pull request merged**
